File: Keeper/Models/VaultKeep.cs
namespace Keeper.Models;

public class VaultKeep : ICreated, IRepoItem<int>
{
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public int VaultId { get; set; }
  public Vault Vault { get; set; }
  public int KeepId { get; set; }
  public Keep Keep { get; set; }
}

File: Keeper.client/src/models/VaultKeep.js
import { Keep } from "./Keep.js"
import { Profile } from "./Profile.js"
import { Vault } from "./Vault.js"

export class VaultKeep {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.views = data.views
    this.kept = data.kept
    this.vaultKeepId = data.vaultKeepId
    this.vaultKeepCreatorId = data.vaultKeepCreatorId
  }
}